src/main/java/app/api/InformationController.java

```diff
@@ -53,9 +68,76 @@ public String getLastBlock() {
                 .limit(1).iterator();
 
         return cursor.hasNext() ? cursor.next().toJson() : "{}";
-
     }
 
+   	@RequestMapping(value = ApiPaths.PRODUCER_YIELD, method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public String getProducerYield() {
+ 
+    	final Integer blocksPerHour = 7200;
+    	final Long witnessNum = 21L;
+       	final Float maxBlocksPerHour = blocksPerHour/ (witnessNum *1.0f);
+   		String localProducer = "";
+    	Long producedBlocks = 0L;
+    	
+    	try {
+    	
+            final MongoCollection<Document> collection = mongoClient.getDatabase("application")
+            		.getCollection("data");
+
+            if(collection.countDocuments() > 0L) {
+            	Document applicationData = collection.find().first();
+            	if(! applicationData.getString("localNodePrivKey").equals("")) {
+            		ECPrivateKey privKey = cryptoService.getECPrivateKeyFromRawString((String) applicationData.getString("localNodePrivKey"));
+   	   				localProducer = cpxkey.getPublicAddressCPX(privKey).toString();
+            	}
+            }
+                 		
+    		final HashMap<String, Long> producerStats = new HashMap<>();
+    		final MongoCursor<Document> cursorMiner = mongoClient.getDatabase("apex")
+                    .getCollection("miner")
+                    .find().sort(new Document("addr", -1))
+                    .limit(0).iterator();
+    		
+    		while (cursorMiner.hasNext()) {
+                producerStats.put((String) cursorMiner.next().get("addr"),0L);
+    		}
+
+            final MongoCursor<Document> cursor = mongoClient.getDatabase("apex")
+               		 .getCollection("block")
+               		 .find(gte("timeStamp", new BsonDateTime(Instant.now().toEpochMilli() - 3600000L)))   
+               		 .limit(0)
+               		 .iterator();
+            
+            if(cursor.hasNext()) {
```

Comment:
Dont need this check if cursor does not have a next Element the forEach does not run anyway